pkgs/collection/lib/src/boollist.dart

```diff
@@ -175,12 +178,16 @@ abstract /*mixin*/ class BoolList with ListMixin<bool> {
     }
   }
 
+  /// Note: [index] is NOT checked for validity.
+  bool _getBit(int index) =>
```

Comment:
FYI: seems far too complex to not centralize. Concerns about not being inlined @lrhn ?

Reply:
Not sure what "centralize" means here - if anything this is centralized in that all bit-reads use this one implementation. 
I hope it's inlined, but I leave that to compilers. They (probably) know better.

For performance, it's likely a little inefficient because computed shifts are expensive on x64 chips.
But everybody does it this way, so hopefully it'll be fine.